### ISSUE TITLE
App on ios emulator not detecting correct emulator to use

### ISSUE THREAD

**Issue by iwiznia** [Expensify/App#12060]
Context https://expensify.slack.com/archives/C03TQ48KC/p1666362417121579

# Problem

Apparently when trying to run the ios app, it is not finding correctly the latest emulator I have installed and for some reason is looking for iPhone13, when I only have iPhone14 installed.

Error:
```
➜  App git:(main) npm run ios

> new.expensify@1.2.18-4 ios
> scripts/set-pusher-suffix.sh && npx react-native run-ios --port=8082

Using PUSHER_DEV_SUFFIX from ../Web-Expensify/_config.local.php
Found suffix: 65b74ba46c7c4edfaf0d7dc4b6c4caf6
Updating .env
warn Package react-native-safe-area-context contains invalid configuration: "dependency.platforms.android.cmakeListsPath" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
warn Package react-native-flipper contains invalid configuration: "dependency.platforms.ios.project" is not allowed. Please verify it's properly linked using "react-native config" command and contact the package maintainers about this.
info Found Xcode workspace "NewExpensify.xcworkspace"
error No simulator available with name "iPhone 13".
info Run CLI with --verbose flag for more details.
```

My emulators:
```
info Run CLI with --verbose flag for more details.
➜  App git:(main) xcrun simctl list devices
== Devices ==
-- iOS 16.0 --
    iPhone SE (3rd generation) (1880E442-085D-45CA-B198-F59D19EC0919) (Shutdown)
    iPhone 14 (E3112A94-F1BA-4B56-9B0F-59B9E48E0546) (Shutdown)
    iPhone 14 Plus (E3537629-E979-4D64-A9D6-D0581584FE98) (Shutdown)
    iPhone 14 Pro (5B1AF5C8-23D9-43AC-8499-2A6B251513DB) (Shutdown)
    iPhone 14 Pro Max (5F4913AB-1821-4C74-8C09-9D5B40755356) (Shutdown)
    iPad (9th generation) (7761FF05-80C3-488A-9638-02AB93151AC8) (Shutdown)
    iPad Pro (11-inch) (3rd generation) (EB1FC844-8E82-48F8-9335-A4F3DE2DE162) (Shutdown)
    iPad Pro (12.9-inch) (5th generation) (53EF422D-533C-4D2A-BEF1-65051AEB9465) (Shutdown)
    iPad Air (5th generation) (3A8C0CCC-B5EF-4F42-93B5-4CCC85D0F3C8) (Shutdown)
    iPad mini (6th generation) (8B66D73F-F99F-4AD1-B354-563A36337D7A) (Shutdown)
```

Workaround: run `npx react-native run-ios --simulator='iPhone 14'` instead of `npm run ios`

**Comment by mountiny** [comment]
This could be External right?

**Comment by Puneet-here** [comment]
This happens because at `node_modules/@react-native-community/cli-platform-ios/build/commands/runIOS/index.js` the default is iPhone 13. You can set any emulator here
<img width="409" alt="Screenshot 2022-10-21 at 9 23 46 PM" src="https://user-images.githubusercontent.com/101883770/197239777-8a96f24f-2d46-4018-ab34-12c8047c006b.png">

**Comment by mountiny** [comment]
> node_modules/@react-native-community/cli-platform-ios/build/commands/runIOS/index.js

Would that not be overwritten when someone runs npm install since it is in noce_modules? How can we make it default?

**Comment by Puneet-here** [comment]
Just tried npm install and it didn't change back to the previous value.

**Comment by Puneet-here** [comment]
Yeah that makes sense but I believe there's no other way to set the default simulator. The only other way to use specific simulator is to use `--simulator` flag but that's also a problem because we can't tell which emulator is available on the system.

**Comment by JmillsExpensify** [comment]
Is this a problem still then? @iwiznia We need to get in the habit of creating issues and ensuring that people are assigned so that these issues don't end up hanging out for months. Accordingly, I'm going to assign you and you can either close / make sure someone is assigned.

**Comment by mountiny** [comment]
This is still a problem I think since everyone should be on XCode 16 and the default one is iPhone 14 we should set that as default so we can update the package.json to use the --simulator flag and use the iPhone 14 right?

**Comment by Puneet-here** [comment]
Yes, I think it makes sense since most people will be using the latest XCode.

We just have to use the line below
`  "ios": "npx react-native run-ios --simulator=\"iPhone 14\"",`
https://github.com/Expensify/App/blob/c638000bce780372f046886db9bc182135973d6a/package.json#L13

**Comment by mountiny** [comment]
@parasharrajat No need to review this, I think Puneet's proposal is good, but it would be great if you could help to test the PR later.

@Puneet-here thank you, please raise a PR with these changes

**Comment by parasharrajat** [comment]
`npx react-native run-ios --port=8082 --simulator=\"iPhone 14\` I am not in favor of doing this. I use iphone 13 pro for testing. IMO, this is more a user side overhead. A user should specify the device they want to use.

**Comment by mountiny** [comment]
> I use iphone 13 pro for testing.

Why this specifically? It would be better to use the standard which is the default iPhone 14, but I am happy with your suggestion.

Although we would break this for everyone since they would need to update the `env` for that.  With setting iphone 14 most people should be fine as up to date xcode uses that

**Comment by parasharrajat** [comment]
my Preference is to cover most devices. There were more iPhone 13 users than 14 so I used that device to test. But it is not specific. Sometimes, I have to test different models based on the problem. 

I think if `$DEVICE` is not set `--simulator=$DEVICE` will be ignored and default will be used. But that needs to be tested first.

**Comment by mountiny** [comment]
Alrighty, I am happy with that solution as well, the goal is for this to work out of the box.

**Comment by parasharrajat** [comment]
OK, great. Let's see what works well on the PR.

**Comment by mdneyazahmad** [comment]
We may not need to hardcode the simulator option. iPhone 14 is added as a preferred fallback simulator in https://github.com/react-native-community/cli/releases/tag/v9.1.2.

React Native Expensify

```
    "@react-native-community/cli": "^9.0.0",
    "@react-native-community/cli-platform-android": "^9.0.0",
    "@react-native-community/cli-platform-ios": "^9.0.0",
```

https://github.com/Expensify/react-native/blob/6e9d4bf1b41242d582c03be5b933b340a832aacf/package.json#L107-L109

--- 

React Native upstream

```
    "@react-native-community/cli": "9.2.1",
    "@react-native-community/cli-platform-android": "9.2.1",
    "@react-native-community/cli-platform-ios": "9.2.1",
```

https://github.com/facebook/react-native/blob/428feb2f76c55d6ae5914b019095cd835aa514b0/package.json#L112-L114

The latest version for `@react-native-community/cli-platform-ios` is v9.2.1. I believe, upgrading the package/packages to v9.1.2+ would fix the issue. Thanks!

**Comment by mountiny** [comment]
Oh good shout @mdneyazahmad, let me have a look into where we are with RN updates

**Comment by parasharrajat** [comment]
What version of RN is required for that?

**Comment by mountiny** [comment]
@Puneet-here @parasharrajat lets do that, I chatted with @roryabraham and this should be fine, we just need to check if everything else works fine with the new RN version.

**Comment by Puneet-here** [comment]
Just to be sure, we wanna use `npx react-native run-ios --port=8082 --simulator=$DEVICE` right? Or are we planning to upgrade RN version.

**Comment by parasharrajat** [comment]
>  are we planning to upgrade RN version

I think.

**Comment by Puneet-here** [comment]
Then I think we should assign @mdneyazahmad here since they researched and pointed out it.

**Comment by parasharrajat** [comment]
@mountiny Can you please confirm what is the expected behavior now?

**Comment by mountiny** [comment]
The expected behaviour is that if you are on up-to-date mac with up-to-date XCode, you can just run `npm run ios` and it works ™️.

I agree that if updating react native (which we will do eitherway) will accomplish this, then we should do this.

in that case @mdneyazahmad would be the best solution, thanks for the message @Puneet-here 

@mdneyazahmad would you be able to update our version of the RN for to the necessary version and test the App with this version? hopefully no breaking changes.

**Comment by mountiny** [comment]
Updating the RN form is kind of a PITA and I havent had time for this yet. Since that takes couple of hours of work and this does not seem like as such a valuable reason to switch I would go ahead with using the original proposal from @Puneet-here.

I hope that make sense, we are trying to squash all the bug and this can be clean up once we eventually update the RN version for something more important I would say.

@Puneet-here Could you please implement the solution you have proposed which will solve this issue quickly and eventually this can be removed once RN will be updated?

**Comment by mountiny** [comment]
Oh Dang, we have updated this already 😂 

In that case this should be fixed now! I think we are good just close this issue then. Not sure if we would pay out anything since there was not really any testing involved here.

Thank you for help everyone though!
